java-squid/src/main/java/org/sonar/java/se/ConstraintManager.java

```diff
@@ -27,10 +27,13 @@
 public class ConstraintManager {
 
   private int counter = ProgramState.EMPTY_STATE.constraintsSize();
-  private SymbolicValue wrappedValue;
+  private SymbolicValueFactory symbolicValueFactory;
 
-  public void setWrappedValue(SymbolicValue wrappedValue) {
-    this.wrappedValue = wrappedValue;
+  public void setValueFactory(SymbolicValueFactory valueFactory) {
+    if (symbolicValueFactory != null) {
```

Comment:
You can use guava preconditions check in this case : https://google-collections.googlecode.com/svn/trunk/javadoc/com/google/common/base/Preconditions.html#checkState(boolean) 

Simplifies a bit the code.